api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -547,6 +561,33 @@ public DomainValuePair apply(DataSetValues dataSetValue) {
     return ResponseEntity.ok(TO_CLIENT_DATA_SET.apply(dbDataSet));
   }
 
+  // If the dataSet has concept set of Domain Person, check if such a concept set exist in workspace
+  // if not create a dummy Concept set with name Demographics and Domain PERSON.
+  private void handleDemographicConceptSet(
+      DataSetRequest request, long workspaceId, Timestamp now) {
+    if (request.getConceptSetIds().size() > 0 && request.getConceptSetIds().contains(-1l)) {
+      request.getConceptSetIds().remove(-1l);
+      // -1 Represent Demographics concepts set, Check for GENDER OR RACE OR ETHNICITY DOMAIN
+      // concept
+      // and add it as a dummy concept under the Demographics concept set
+      List<Concept> genderRaceConcept = conceptDao.findGenderRaceEthnicityFromConcept();
+      if (genderRaceConcept.size() > 0) {
+        org.pmiops.workbench.db.model.ConceptSet demographicsCS =
+            new org.pmiops.workbench.db.model.ConceptSet();
+        demographicsCS.setName("Demographics");
```

Comment:
What happens if a user has a concept set named "Demographics"?

Reply:
Yes this will restrict user to have a concept set named Demographics. 

However i am hoping this will go away once and if we allow user to add/have Demographic Domain concept sets under the concept set page.

Reply:
Okay, we should make sure the messaging on that is reasonable. And block the concept set creation API from allowing that to be submitted for now.